src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/measure/ActionMeterTest.kt

```diff
@@ -158,6 +160,39 @@ class ActionMeterTest {
         )
     }
 
+    @Test
+    fun shouldHookAction() {
+        val output = CollectionActionMetricOutput(mutableListOf())
+        val countingPostMeasurementAction = CountingPostMeasurementAction()
+        val tick = ofSeconds(1)
+        val clock = TickingClock(start, tick)
+        val actionMeter = ActionMeter.Builder(
+            output = output
+        )
+            .virtualUser(vu)
+            .clock(clock)
+            .postMeasurementAction(countingPostMeasurementAction)
+            .build()
+
+        actionMeter.measure(CREATE_ISSUE, clock::tick)
+        actionMeter.measure(VIEW_BOARD, clock::tick)
+
+        assertThat(countingPostMeasurementAction.getCount()).isEqualTo(2)
+        assertThat(output.metrics.map { it.duration }).hasOnlyOneElementSatisfying { ofSeconds(1) }
+    }
+
+    private class CountingPostMeasurementAction : PostMeasurementAction {
+        private val counter = AtomicInteger()
+        override fun run(webDriver: RemoteWebDriver) {
```

Comment:
Perhaps we could generalize `W3cPerformanceTimeline` as a hook later. That one needs to know the `ActionMetric` (or `ActionMetric.Builder`) to make decisions. It also needs the `ActionMetric.Builder` to contribute `drilldown` to it.
These are dynamic and cannot be passed via hook constructor.

But e.g. `ScreenshotHook` could inject `RemoteWebDriver` (or better `TakesScreenshot`) via its constructor no problem. So we don't have to pass them via hook API. Especially since `ActionMeter` has no idea about WD anyways.